internal/hcl/parser.go

```diff
@@ -189,8 +202,17 @@ func (p *Parser) ParseDirectory() (*Module, error) {
 		nil,
 		p.workspaceName,
 		p.blockBuilder,
+		p.newSpinner,
 	)
 
+	if v := evaluator.MissingVars(); len(v) > 0 {
+		log.Warnf(
+			"Input values were not provided for the following Terraform variables: %s",
+			strings.TrimRight(strings.Join(v, ", "), ", "),
+		)
+		log.Warnf("Try using --terraform-var-file or --terraform-var to specify input vars")
```

Comment:
ah ok, yeah I'll alter

Reply:
resolved in https://github.com/infracost/infracost/pull/1554/commits/c207cfb94ade593bcec987a2998c44506e547e53, however because of the way we do output buffering this means that the warning shows inbetween spinners:

```
 ✔ Downloading Terraform modules
  Warning: Input values were not provided for the following Terraform variables: 'variable.foo', 'variable.baz'. Use --terraform-var-file or --terraform-var to specify input vars.
  ✔ Evaluating Terraform directory
  ✔ Retrieving cloud prices to calculate costs
 ```
 
 There's no easy fix for this as the warning happens at a lower level than the last spinner (cloud prices) so it would mean having a variable passed around the already messy run files. Are we ok with the messaging as is? cc @aliscott

Reply:
@hugorut thanks! That looks fine - no need for messy workaround.